src/pages/[...slug].tsx

```diff
@@ -1,110 +1,75 @@
-import {
-  GetStaticPathsContext,
-  GetStaticPathsResult,
-  GetStaticPropsContext,
-  GetStaticPropsResult,
-} from 'next'
-import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
-import { drupalClient } from '@/utils/drupalClient'
-import { entityMeta } from '@/lib/delegators/entityMetaProvider'
-import { NodeTypes } from '@/types/node'
-import { getGlobalElements } from '@/lib/context/getGlobalElements'
-import Layout from '@/components/layout'
-import { generalEntityDataService } from '@/lib/delegators/generalEntityDataService'
+import * as React from "react"
+import { GetStaticPathsResult } from "next"
 
-interface EntityProps {
-  entityProps: any
+import { drupalClient } from "@/utils/drupalClient"
+import { queries } from "@/data/queries"
+import { NewsStoryPageProps } from "../types"
+import Layout from "@/components/layout"
+import { NewsStoryFull } from "@/components/news_story"
+
+const RESOURCE_TYPES = ["node--news_story"] as const
+
+interface ResourcePageProps {
+  resource: NewsStoryPageProps
 }
 
-/** Return the layout and primary component and props. */
-export default function Page({ entityProps, props }) {
-  const Component = entityMeta[entityProps.type].component
+export default function ResourcePage({ resource }: ResourcePageProps) {
+  if (!resource) return null
 
   return (
-    <Layout props={props}>
-      <Component {...entityProps} />
+    <Layout>
+      {resource.type === "node--news_story" && <NewsStoryFull {...resource} />}
     </Layout>
   )
 }
 
-/** All active node types are identified by the keys of the collected node meta info. */
-const resourceTypes = Object.keys(entityMeta)
-
-export async function getStaticPaths(
-  context: GetStaticPathsContext
-): Promise<GetStaticPathsResult> {
+export async function getStaticPaths(context): Promise<GetStaticPathsResult> {
   return {
     paths: await drupalClient.getStaticPathsFromContext(
-      resourceTypes,
-      context,
-      {
-        params: {},
-      }
+      Array.from(RESOURCE_TYPES),
+      context
     ),
-    fallback: 'blocking',
+    fallback: "blocking",
   }
 }
 
-/** @todo This cannot handle non-node urls yet. */
-export async function getStaticProps(
-  context: GetStaticPropsContext
-): Promise<GetStaticPropsResult<EntityProps>> {
-  const params = new DrupalJsonApiParams()
+export async function getStaticProps(context) {
   const path = await drupalClient.translatePathFromContext(context)
 
-  if (!path || !resourceTypes.includes(path.jsonapi.resourceName)) {
+  if (!path) {
```

Comment:
I like our version better here.

Reply:
would we still want to have additional type checks in any spot? i.e. at the start of the mapping functions?